stock_picking_report_valued_sale_mrp/models/stock_move_line.py

```diff
@@ -37,10 +37,18 @@ def _compute_sale_order_line_fields(self):
         amounts according to the corresponding delivered kits"""
         super()._compute_sale_order_line_fields()
         pickings = self.mapped("picking_id")
-        kit_lines = pickings.move_line_ids.filtered("phantom_product_id")
         pickings.move_line_ids.write(
             {"phantom_line": False, "phantom_delivered_qty": 0.0}
         )
+        for picking in pickings:
+            self.filtered(
+                lambda x: x.picking_id == picking
+            )._compute_sale_order_line_fields_by_picking()
+
+    def _compute_sale_order_line_fields_by_picking(self):
+        """We want to compute the lines value by picking to avoid mixing lines
+        if they wen't altogether"""
```

Comment:
Not sure if this is what you want to say:

```suggestion
        if they weren't shipped altogether.
        """
```